src/CMakeLists.txt

```diff
@@ -25,6 +25,9 @@ target_link_libraries(Sc2LadderCore
     sc2api sc2lib sc2utils sc2protocol civetweb libprotobuf
 )
 
+if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
+    target_link_libraries(Sc2LadderCore stdc++fs)
```

Comment:
This might mean that OS X is broken :)

If you want to check the compiler, this macros could be useful: `CMAKE_COMPILER_IS_GNUCXX`.
But probably we should do smthg like:
```
if (UNIX)
???
```

Reply:
Regarding the compiler macros see:
https://github.com/alkurbatov/suvorov-bot/blob/master/src/CMakeLists.txt#L25

Reply:
Whether it is broken it not is the question. In the threads I have read it always said: with gcc(!) you need to set that compiler flag. Can you test it with clang? I don't want to introduce unnecessary flags.

Reply:
Thank you.

Reply:
```
[ 95%] Building CXX object src/CMakeFiles/Sc2LadderCore.dir/sc2laddercore/AgentsConfig.cpp.o
/Users/alkurbatov/work/tmp/Sc2LadderServer/src/sc2laddercore/AgentsConfig.cpp:12:10: fatal error: 'experimental/filesystem' file not found
#include <experimental/filesystem>
```

Reply:
I've googled a bit. Looks like the only way for High Sierra is to instal g++ :)
I will update to Mojave soon and recheck, but it will take some time.

Reply:
Unfortunately, the same on Mojave.

Reply:
I'll look at ways to remove it, we might just have to use platform specific functions instead